javascript/javascript1/week2/flight_booking.js

```diff
@@ -0,0 +1,29 @@
+function getFullname(firstname,surname,useFormalName,gender){
+    if(firstname===' ' || surname===' '){
+        console.log('Enter the valid name ?');
+  // prompt('enter the valid firstname and surname');
+    }
+    if(useFormalName){
+        if(gender==='male'){
+            return `Lord ${firstname} ${surname}`;
+        }else if(gender==='female'){
+            return `Lady ${firstname} ${surname}`;
+        }else{
+            return `Other ${firstname} ${surname}`;
+        }
+    }else{
+        return `${firstname} ${surname}`;
+    }
```

Comment:
Here you can reduce the amount of nesting (`if` inside `if`):
```js
    if(useFormalName){
        return `${firstname} ${surname}`;
    }

    if(gender==='male'){
        return `Lord ${firstname} ${surname}`;
    }else if(gender==='female'){
        return `Lady ${firstname} ${surname}`;
    }

    return `Other ${firstname} ${surname}`;
```

IMO this makes the function easier to read.

Reply:
Ups I have a bug :point_up: 

The following should have been inverted (notice the `!`):
```js
    if(!useFormalName){
        return `${firstname} ${surname}`;
    }

    ...
```